src/storage/write_ahead_log/log_serializer_task.cpp

```diff
@@ -112,12 +112,16 @@ void LogSerializerTask::SerializeRecord(const terrier::storage::LogRecord &recor
       // ProjectedRowInitializer from these ids and their corresponding block layout.
       WriteValue(delta->NumColumns());
       WriteValue(delta->ColumnIds(), static_cast<uint32_t>(sizeof(col_id_t)) * delta->NumColumns());
-
+      // Write out the attr sizes. We write these out because there is no guarantee we have the block layout during
+      // deserialization
+      const auto &block_layout = record_body->GetTupleSlot().GetBlock()->data_table_->GetBlockLayout();
+      for (uint16_t i = 0; i < delta->NumColumns(); i++) {
+        WriteValue(block_layout.AttrSize(delta->ColumnIds()[i]));
```

Comment:
This was simpler because we inline varlens, and I didn't understand the logic in how we determine the boundaries in the projected columns code. I'll see if I can understand it more and do the same.

Reply:
This was easier, and honestly I didn't understand the code that generates the attribute boundaries. I think we can actually get away with just writing out how many attributes of each size we have.